### PR TITLE
Man 9906 terraform provider update cluster bug

### PR DESCRIPTION
**Pull request by bensagi**

### CODE REVIEW

File: securecn/resource_portshift_k8s_cluster.go [hunk 1]
```diff
@@ -557,6 +546,15 @@ func getClusterFromConfig(d *schema.ResourceData) (*model.KubernetesCluster, err
 		}
 	}
 
+	url := utils2.ReadNestedStringFromTF(d, InternalRegistryFieldName, "url", 0)
```
Comment (rafaelseidelcisco) [inline securecn/resource_portshift_k8s_cluster.go:540]: const string?

File: securecn/resource_portshift_k8s_cluster.go [hunk 2]
```diff
@@ -557,6 +546,15 @@ func getClusterFromConfig(d *schema.ResourceData) (*model.KubernetesCluster, err
...
+	if url != "" {
```
Comment (rafaelseidelcisco) [inline securecn/resource_portshift_k8s_cluster.go:541]: can there be an error in line 549?

File: securecn/resource_portshift_k8s_cluster.go [hunk 3]
```diff
@@ -344,20 +324,28 @@ func resourceClusterDelete(ctx context.Context, d *schema.ResourceData, m interf
...
-		err = utils2.MakeExecutable(vaultCertsGenFilePath)
+		err = utils2.MakeExecutable(installationDir + "/" + vaultCertsGenFilePath)
 		if err != nil {
 			return err
 		}
 	}
 
 	if tracingEnabled {
-		err = utils2.MakeExecutable(tracingCertsFilePath)
+		err = utils2.MakeExecutable(installationDir + "/" + tracingCertsFilePath)
```
Comment (rafaelseidelcisco) [inline securecn/resource_portshift_k8s_cluster.go:348]: in theory the slash should be OS agnostic (it'l be different in windows)

File: securecn/resource_portshift_k8s_cluster.go [hunk 4]
```diff
@@ -628,6 +626,26 @@ func updateMutableFields(d *schema.ResourceData, secureCNCluster *model.Kubernet
...
+	_ = d.Set(RestrictRegistriesFieldName, secureCNCluster.RestrictRegistires)
+	_ = d.Set(IstioIngressEnabledFieldName, secureCNCluster.IsIstioIngressEnabled)
+	_ = d.Set(IstioIngressAnnotationsFieldName, secureCNCluster.IstioIngressAnnotations)
+	_ = d.Set(EnableApiIntelligenceDASTFieldName, secureCNCluster.APIIntelligenceDAST)
+	_ = d.Set(EnableAutoLabelFieldName, secureCNCluster.AutoLabelEnabled)
+	_ = d.Set(HoldApplicationUntilProxyStartsFieldName, secureCNCluster.IsHoldApplicationUntilProxyStarts)
+	_ = d.Set(InternalRegistryFieldName, utils2.GetTfMapFromKeyValuePairs([]utils2.KeyValue{{"url", secureCNCluster.InternalRegistryParameters.InternalRegistry}}))
+	_ = d.Set(ExternalCAFieldName, utils2.GetTfMapFromKeyValuePairs([]utils2.KeyValue{{"id", secureCNCluster.ExternalCa.ID}, {"name", secureCNCluster.ExternalCa.Name}}))
+	_ = d.Set(SidecarResourcesFieldName, utils2.GetTfMapFromKeyValuePairs([]utils2.KeyValue{
+		{"proxy_init_limits_cpu", secureCNCluster.SidecarsResources.ProxyInitLimitsCPU},
```
[the remark this answers is not in the document]
Comment (bensagi) [reply securecn/resource_portshift_k8s_cluster.go:634]: not exactly. I changed to consts

**Pull request merged**